### ISSUE TITLE
MayBeConst false negative when concatenating strings

### ISSUE THREAD

**Issue by vanniktech** [arturbosch/detekt#900]
```kotlin
private const val PREF_SUFFIX = "gtar-"
private val KEY_LAST_SHOWN = PREF_SUFFIX + "last_shown"
```

In this case `KEY_LAST_SHOWN` won't get flagged in the `MayBeConst` rule although it can and should be a const.

**Comment by 3flex** [comment]
I think this needs symbol resolution? Playing in IntelliJ an error is shown when PREF_SUFFIX isn't a const val, so the detekt rule would have to check that any variable in an expression is a const val. Symbol resolution is needed to check that.

**Comment by arturbosch** [comment]
I would not say this is a bug, rather an improvement.

**Comment by schalkms** [comment]
Closed with 2025fe5
